pkg/apis/operator/v1alpha1/tektonconfig_types.go

```diff
@@ -49,6 +49,12 @@ func (tp *TektonConfig) GetStatus() TektonComponentStatus {
 
 // Prune defines the pruner
 type Prune struct {
+	// image used in cron job
+	// +optional
+	Image string `json:"image,omitempty"`
+	// apply the prune job to the individual resources
+	// +optional
+	ApplyToIndividualResource bool `json:"apply-to-individual-resource,omitempty"`
```

Comment:
this is to apply the values of keep and keep-since at the individual resource level like if set to true, it will keep those many pipelineruns/taskruns for that pipeline/task. I think this should be good to have,  but any other name suggestions cc @vdemeester @concaf 

we will need docs to be added for this

Reply:
I think we will need the default values also added for this as `false`

Reply:
@vdemeester @piyush-garg  How about 
* `KeepByResource` => `keep-by-resource`
* `LimitByResource` => `limit-by-resource`
* `FilterByResource` => `filter-by-resource`

I am poor at naming, please suggest a name :slightly_smiling_face: 

Note: we are limiting the number of resources with `keep` parameter

Reply:
I think `prune-per-resource` is better

Reply:
@piyush-garg updated